test/e2e/storage/volume_provisioning.go

```diff
@@ -1016,6 +1163,32 @@ func getDefaultPluginName() string {
 	return ""
 }
 
+func addAllowedTopologiesToStorageClass(c clientset.Interface, sc *storage.StorageClass) sets.String {
+	zones, err := framework.GetClusterZones(c)
+	Expect(err).ToNot(HaveOccurred())
+	zoneCount := rand.Intn(zones.Len())
```

Comment:
what happens if zones.Len() == 0?

instead of randomly picking a number of zones, I would make the test deterministic and have a test for 1 zone for now.  Maybe in the future, we can have a test case for multiple zones, and we can deterministically control which zones to use (ie through pod node selector or something)

Reply:
True .. AllowedTopology is ignored during provisioning so we can set it to any random zone and the test should still succeed. So (3) is unnecessary above.